src/modules/coreHttpApi/controllers/AttributesController.ts

```diff
@@ -74,6 +74,96 @@ export class AttributesController extends BaseController {
+
+        const query: Record<string, any> = {};
+
+        Object.entries(urlQuery).forEach(([key, value]) => {
+            if (key.startsWith("query.")) {
+                query[key.replace("query.", "")] = typeof value === "string" ? value : "";
+            }
+        });
+
+        const result = await this.consumptionServices.attributes.getOwnSharedAttributes({
+            peer: typeof urlQuery.peer === "string" ? urlQuery.peer : "",
+            hideTechnical: typeof urlQuery.hideTechnical === "string" ? urlQuery.hideTechnical === "true" : false,
+            query: query,
+            onlyLatestVersions: typeof urlQuery.onlyLatestVersions === "string" ? urlQuery.onlyLatestVersions === "true" : false,
+            onlyValid: typeof urlQuery.onlyValid === "string" ? urlQuery.onlyValid === "true" : false
+        });
+        return this.ok(result);
+    }
+
+    @GET
+    @Path("/Peer/Shared/Identity")
+    @Accept("application/json")
+    public async getPeerSharedIdentityAttributes(@Context context: ServiceContext): Promise<Envelope> {
+        const urlQuery = context.request.query;
+
+        const query: Record<string, any> = {};
+
+        Object.entries(urlQuery).forEach(([key, value]) => {
+            if (key.startsWith("query.")) {
+                query[key.replace("query.", "")] = typeof value === "string" ? value : "";
+            }
+        });
+
+        const result = await this.consumptionServices.attributes.getPeerSharedAttributes({
+            peer: typeof urlQuery.peer === "string" ? urlQuery.peer : "",
+            hideTechnical: typeof urlQuery.hideTechnical === "string" ? urlQuery.hideTechnical === "true" : false,
+            query: query,
+            onlyLatestVersions: typeof urlQuery.onlyLatestVersions === "string" ? urlQuery.onlyLatestVersions === "true" : false,
+            onlyValid: typeof urlQuery.onlyValid === "string" ? urlQuery.onlyValid === "true" : false
+        });
+        return this.ok(result);
+    }
+
+    @GET
+    @Path("/:id/Versions")
+    @Accept("application/json")
+    public async getVersionsOfAttribute(@PathParam("id") attributeId: string): Promise<Envelope> {
+        const result = await this.consumptionServices.attributes.getVersionsOfAttribute({
+            attributeId
+        });
+        return this.ok(result);
+    }
+
+    @GET
+    @Path("/:id/Versions/Shared")
+    @Accept("application/json")
+    public async getSharedVersionsOfRepositoryAttribute(
+        @PathParam("id") attributeId: string,
+        @QueryParam("peers") peers?: string | string[],
+        @QueryParam("onlyLatestVersion") onlyLatestVersion?: string
+    ): Promise<Envelope> {
+        if (typeof peers === "string") {
+            peers = [peers];
+        }
+
+        const result = await this.consumptionServices.attributes.getSharedVersionsOfRepositoryAttribute({
+            attributeId,
+            onlyLatestVersions: onlyLatestVersion ? onlyLatestVersion === "true" : undefined,
```

Comment:
```suggestion
            onlyLatestVersions: !!onlyLatestVersion,
```

Reply:
Sadly also not possible as the boolean pa rams come in as string and therefor true and false would be true.

Also, the default is true (Not yet clear if intended will be discues with @Milena-Czierlinski), that's why we need to pass undefined if not present